pkg/agent/flowexporter/connections/connections.go

```diff
@@ -234,14 +256,22 @@ func (cs *connectionStore) ForAllConnectionsDo(callback flowexporter.ConnectionM
 func (cs *connectionStore) Poll() ([]int, error) {
 	klog.V(2).Infof("Polling conntrack")
 	// Reset isActive flag for all connections in connection map before dumping flows in conntrack module.
-	// This is to specify that the connection and the flow record can be deleted after the next export.
-	resetConn := func(key flowexporter.ConnectionKey, conn flowexporter.Connection) error {
-		conn.IsPresent = false
-		cs.connections[key] = conn
+	// if the connection does not exist in conntrack table and exported, we will delete it from connection map.
+	deleteIfStaleOrResetConn := func(key flowexporter.ConnectionKey, conn flowexporter.Connection) error {
+		if !conn.IsPresent && conn.DoneExport {
+			if err := cs.DeleteConnectionByKeyWithoutLock(key); err != nil {
```

Comment:
Since it is only done in connections.go and only at one place, wondering if we need a method in the interface. We also do not expose the lock to the public? The only way it can be used is with `ForAllConnections`. 
We can either delete it or make it the method of the struct. Any thoughts?

Reply:
Makes sense to me. Removed it from interface.